src/main/java/core/storage/Storage.java

```diff
@@ -0,0 +1,95 @@
+package core.storage;
+
+import core.exceptions.NoFruitsEnoughException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class Storage implements FruitOperations {
```

Comment:
```suggestion
public class StorageService {
```

let's have in this class only `addToStorage()`, `removeFromStorage()` and `getRemainingFruit()` methods